external/solr/src/main/java/org/apache/stormcrawler/solr/persistence/SolrSpout.java

```diff
@@ -135,7 +157,8 @@ else if (resetFetchDateAfterNSecs != -1) {
 
         query.setQuery("*:*")
                 .addFilterQuery("nextFetchDate:[* TO " + lastNextFetchDate + "]")
-                .setSort("nextFetchDate", ORDER.asc);
+                .setSort("nextFetchDate", ORDER.asc)
+                .setParam("shards", "shard" + shardID);
```

Comment:
IIRC the way it works with the other backends is that if the number of shards in the conf is either unspecified or 1, we let a single instance take care of everything and so this line
```
  .setParam("shards", "shard" + shardID);
```
would not be executed

Reply:
The idea was to have the `shardID` default to 1 when the respective property is not set.

- [Definition](https://github.com/apache/incubator-stormcrawler/pull/1343/files#diff-a499db920515864e856bb6616ff489daf6a58af1159b2f6057ea7da2273e774eR64)
- [Getting the number of shards from the conf](https://github.com/apache/incubator-stormcrawler/pull/1343/files#diff-a499db920515864e856bb6616ff489daf6a58af1159b2f6057ea7da2273e774eR88-R94)

The request to Solr will then work since the collection will have just one shard: `shard1`. I will add a test for this to make sure I'm not missing anything.

Reply:
I added some defaults to the scripts as well. [d22c325](https://github.com/apache/incubator-stormcrawler/pull/1343/commits/d22c3259df1c9ec07b4b3d2c386cf2385ee0bc23)

Reply:
I would simply not set the param constraint in the query if the config for the number of shards is set or defaults to 1, this way it wouldn't matter how the collection is configured - you would get all the results. Doesn't matter too much though and the current behaviour is also fine.